ERCS/erc-7588.md

```diff
@@ -0,0 +1,102 @@
+
+Typically, rollups keep track of their own posted blob transactions, download blobs and decode their layer 2 transaction data from blobs on demand. However, some third party solutions, such as Portal Network, blobscan, etc., may also index all the blobs ever posted to Ethereum and provide querying services of them. It would greatly improve the visibility and auditability of blobs if we could attach some metadata to the blobs, such as `originator`, `description`, `content_type`, etc.
+
+Moreover, some `dStorage` projects may use blob transactions to post user data to Ethereum, sync and permanently store the blobs off-chain for future retrieval. Attaching metadata to blobs would also inspire novel applications, such as inscriptions, etc.
+
+
+## Specification
+
+### Metadata JSON Schema
+
+The metadata is a JSON object that conforms to the following schema:
+
+```json
+{
+    "title": "Blobs Metadata",
+    "type": "object",
+    "properties": {
+        "originator": {
+            "type": "string",
+            "description": "Identifies the originator of the carried blobs"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the contents of the blobs"
+        },
+        "content_type": {
+            "type": "string",
+            "description": "Describes the mime type of the blobs"
+        },
+        "extras": {
+            "type": "string",
+            "description": "Dynamic extra information of the blobs"
+        },
+        "blobs": {
+            "type": "array",
+            "items": {
+                "description": {
+                    "type": "string",
+                    "description": "Describes the content of the i'th blob"
+                },
+                "content_type": {
+                    "type": "string",
+                    "description": "Describes the mime type of the i'th blob"
+                },
+                "extras": {
+                    "type": "string",
+                    "description": "Dynamic extra information of the i'th blob"
+                },
+            }
+        }
+    }
+}
+```
+
+
+### Metadata Encoding
+
+The metadata JSON object should be converted to hexadecimal with UTF-8 character encoding, and put in the `data` field of the blob transaction.
+
+## Rationale
+
+The rationale for this EIP is to provide a standard way for attaching metadata to blobs carried by blob transactions, which can enhance the auditability and usability of blob transactions. The metadata can be used by various applications and services that post or index blob transactions, such as rollups, `dStorage` projects, blobscan, etc. The metadata can also provide additional context and information about the blobs, such as their `originator`, `description`, `content type`, etc.
```

Comment:
The rationale section should explain technical choices made within the EIP itself, while the motivation justifies the EIP as a whole. This paragraph belongs in Motivation.

My favourite analogy for this is:

> Motivation: Why do we need to build a shed?
> Rationale: Why do we need to paint the shed red?

Reply:
Hi Sam, thank you for the insightful comments! I wanted to let you know that I am actively addressing the feedback and will be updating the pull request shortly. :)